pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -378,11 +378,19 @@ def test_setitem_df_wrong_column_number(self, cols):
     def test_setitem_listlike_indexer_duplicate_columns(self):
         # GH#38604
         df = DataFrame([[1, 2, 3]], columns=["a", "b", "b"])
-        rhs = DataFrame([[10, 11, 12]], columns=["d", "e", "c"])
+        rhs = DataFrame([[10, 11, 12]], columns=["a", "b", "b"])
```

Comment:
this seems more obvious.  should we also be testing the ["d", "e", "c"] case, or is that not well-defined?

Reply:
This aligns, so sets the values to nan. There is a discussion to deprecate this behavior (#39373). So did not want to add tests here